sdk/core/Azure.Core/src/MultipartFormDataContent.cs

```diff
@@ -0,0 +1,397 @@
+            }
+
+            // Extra CRLF to end headers (even if there are no headers).
+            scratch.Append(CrLf);
+
+            return scratch.ToString();
+        }
+
+        private static void EncodeStringToStream(Stream stream, string input)
+        {
+            byte[] buffer = Encoding.Default.GetBytes(input);
+            stream.Write(buffer, 0, buffer.Length);
+        }
+
+        private static Task EncodeStringToStreamAsync(Stream stream, string input, CancellationToken cancellationToken)
+        {
+            byte[] buffer = Encoding.Default.GetBytes(input);
+            return stream.WriteAsync(buffer, 0, buffer.Length, cancellationToken);
+        }
+
+        /// <summary>
+        /// Attempts to compute the length of the underlying content, if available.
+        /// </summary>
+        /// <param name="length">The length of the underlying data.</param>
+        public override bool TryComputeLength(out long length)
+        {
+            int boundaryLength = GetEncodedLength(_boundary);
+
+            long currentLength = 0;
+            long internalBoundaryLength = s_crlfLength + s_dashDashLength + boundaryLength + s_crlfLength;
+
+            // Start Boundary.
+            currentLength += s_dashDashLength + boundaryLength + s_crlfLength;
+
+            bool first = true;
+            foreach (MultipartRequestContent content in _nestedContent)
+            {
+                if (first)
+                {
+                    first = false; // First boundary already written.
+                }
+                else
+                {
+                    // Internal Boundary.
+                    currentLength += internalBoundaryLength;
+                }
+
+                // Headers.
+                foreach (KeyValuePair<string, string> headerPair in content._headers)
+                {
+                    currentLength += GetEncodedLength(headerPair.Key) + s_colonSpaceLength;
+                    currentLength += GetEncodedLength(headerPair.Value);
+                    currentLength += s_crlfLength;
+                }
+
+                currentLength += s_crlfLength;
+
+                // Content.
+                if (!content._requestContent.TryComputeLength(out long tempContentLength))
+                {
+                    length = 0;
+                    return false;
+                }
+                currentLength += tempContentLength;
+            }
+
+            // Terminating boundary.
+            currentLength += s_crlfLength + s_dashDashLength + boundaryLength + s_dashDashLength + s_crlfLength;
+
+            length = currentLength;
+            return true;
+        }
+
+        private static int GetEncodedLength(string input)
+        {
+            return Encoding.Default.GetByteCount(input);
+        }
+
+        #endregion Serialization
+
+        private class MultipartRequestContent
+        {
+            public readonly RequestContent _requestContent;
```

Comment:
Use public naming rules
```suggestion
            public readonly RequestContent RequestContent;
```